zipline/assets/asset_writer.py

```diff
@@ -0,0 +1,679 @@
+        # Generate exchanges DataFrame #
+        ################################
+
+        # Default values to be written to database.
+        exchanges_defaults = {
+            'exchange': None,
+            'timezone': None,
+        }
+
+        # The columns to be returned.
+        exchanges_cols = {'exchange', 'timezone', }
+
+        # The columns provided.
+        cols = set(data.exchanges.columns)
+
+        # Drop columns with unrecognised headers.
+        data.exchanges.drop(cols - (cols & exchanges_cols), axis=1,
+                            inplace=True)
+
+        # Get those columns which we need but
+        # for which no data has been supplied.
+        need = exchanges_cols - set(data.exchanges.columns)
+
+        # Combine the users supplied data with our required columns.
+        exchanges_output = pd.concat(
+            (data.exchanges, pd.DataFrame(
+                self.dict_subset(exchanges_defaults, need),
+                data.exchanges.index,
+            )),
+            axis=1,
+            copy=False
+        )
+
+        ###################################
+        # Generate root symbols DataFrame #
+        ###################################
+
+        # Default values to be written to database.
+        root_symbols_defaults = {
+            'root_symbol': None,
+            'sector': None,
+            'description': None,
+            'exchange_id': None,
+        }
+
+        # The columns to be returned.
+        root_symbols_cols = {'root_symbol', 'sector',
+                             'description', 'exchange_id'}
+
+        # The columns provided.
+        cols = set(data.root_symbols.columns)
+
+        # Drop columns with unrecognised headers.
+        data.root_symbols.drop(cols - (cols & root_symbols_cols), axis=1,
+                               inplace=True)
+
+        # Get those columns which we need but
+        # for which no data has been supplied.
+        need = root_symbols_cols - set(data.root_symbols.columns)
+
+        # Combine the users supplied data with our required columns.
+        root_symbols_output = pd.concat(
+            (data.root_symbols, pd.DataFrame(
+                self.dict_subset(root_symbols_defaults, need),
+                data.root_symbols.index,
+            )),
+            axis=1,
+            copy=False
+        )
+
+        return AssetData(equities=equities_output,
+                         futures=futures_output,
+                         exchanges=exchanges_output,
+                         root_symbols=root_symbols_output)
+
+    def convert_datetime(self, dt):
+        """Convert a datetime variable to integer of nanoseconds
+           since UNIX Epoch.
+
+        Parameters
+        ----------
+        dt
+            A string, int or pd.Timestamp instance representing a datetime.
+
+        Returns
+        -------
+        float
+            nanoseconds since UNIX Epoch.
+
+        """
+
+        if pd.isnull(dt):
+            return None
```

Comment:
The docstring above does not mention that this is the case. Should this not just raise some error? Do we need to explicitly guard against this?